### PR TITLE
Final PR at updating some python ports

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.7.1 21G920 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-certomancer-csc-dummy/Portfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-certomancer-csc-dummy
+version             0.2.3
+revision            0
+
+supported_archs     noarch
```
Comment (reneeotten) [inline python/py-certomancer-csc-dummy/Portfile:10]: it this is "noarch" it should potentially also set: "platforms {darwin any}".

File: python/py-certomancer-csc-dummy/Portfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+checksums           rmd160  6f22228ae49d421bc6708b1dd2f54d8f3bc004e4 \
+                    sha256  ea7016ccaed684d876ce03896f04c11fe7f82cd4ad01413862bf989ee63d6e69 \
+                    size    10588
+
+python.versions     310 311 312
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
```
Comment (reneeotten) [inline python/py-certomancer-csc-dummy/Portfile:27]: this is implied with the default "python.pep517 yes"

File: python/py-certomancer/Portfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-certomancer
+version             0.11.0
+revision            0
+
+supported_archs     noarch
```
Comment (reneeotten) [inline python/py-certomancer/Portfile:10]: same as above

File: python/py-certomancer/Portfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+homepage            https://pypi.org/project/certomancer
+
+checksums           rmd160  34007fbd2d2a1865d08899ab598bd92c2b93c1c5 \
+                    sha256  f414455830fde27265b12dc50a7f619a63479597761f96200e082c8a8d3e55b5 \
+                    size    74450
+
+python.versions     310 311 312
+
+python.pep517       yes
```
Comment (reneeotten) [inline python/py-certomancer/Portfile:25]: this is the default for Python 3.7+

File: python/py-hanko-certvalidator/Portfile [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-hanko-certvalidator
+python.rootname     pyhanko-certvalidator
+version             0.26.3
+revision            0
+
+supported_archs     noarch
```
Comment (reneeotten) [inline python/py-hanko-certvalidator/Portfile:11]: same as above

File: python/py-hanko-certvalidator/Portfile [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+homepage            https://pypi.org/project/pyhanko-certvalidator
+
+checksums           rmd160  a2f06306d28e85d5d797f59919d919011622c913 \
+                    sha256  47fba8e9dbf846d766f2e0a453572dd4b25b2f1397847a31fe892c8eb00391f5 \
+                    size    102722
+
+python.versions     310 311 312
+
+python.pep517       yes
```
Comment (reneeotten) [inline python/py-hanko-certvalidator/Portfile:27]: As above, default for Python 3.7+

File: python/py-hanko/Portfile [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-hanko
+python.rootname     pyHanko
+version             0.21.0
+revision            0
+
+supported_archs     noarch
```
Comment (reneeotten) [inline python/py-hanko/Portfile:11]: as above

File: python/py-hanko/Portfile [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+homepage            https://pypi.org/project/pyHanko
+
+checksums           rmd160  cdf3fa857d6929b06eebf84d869f2bba5a9fa407 \
+                    sha256  e5609fb8a0b8d60168999cffaa382f0361066cb429204ad2b76e84e6e7f5eb93 \
+                    size    376386
+
+python.versions     310 311 312
+
+python.pep517       yes
```
Comment (reneeotten) [inline python/py-hanko/Portfile:26]: as above

File: python/py-oscrypto/Portfile [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-oscrypto
+version             1.3.0
+revision            0
+
+supported_archs     noarch
```
Comment (reneeotten) [inline python/py-oscrypto/Portfile:10]: as above

File: python/py-oscrypto/Portfile [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+checksums           rmd160  e09cbacb792be1f4d06952c3ab1951ae32bf571b \
+                    sha256  6f5fef59cb5b3708321db7cca56aed8ad7e662853351e7991fcf60ec606d47a4 \
+                    size    184590
+
+python.versions     310 311 312
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
```
Comment (reneeotten) [inline python/py-oscrypto/Portfile:30]: as above, this is an dependency already added because of "python.pep517 yes"

**Comment by pmetzger** [comment]
CI failed.

File: python/py-python-pae/Portfile [hunk 11]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               python 1.0
+
+name                    py-python-pae
+version                 0.1.0
+revision                0
+
+supported_archs         noarch
```
Comment (reneeotten) [inline python/py-python-pae/Portfile:10]: as above

File: python/py-uritools/Portfile [hunk 12]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-uritools
+version             4.0.2
+revision            0
+
+supported_archs     noarch
```
Comment (reneeotten) [inline python/py-uritools/Portfile:10]: as above

File: python/py-uritools/Portfile [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+checksums           rmd160  a55def1e60764c671a1cea7c3fd87ebaee68dac0 \
+                    sha256  04df2b787d0eb76200e8319382a03562fbfe4741fd66c15506b08d3b8211d573 \
+                    size    23673
+
+python.versions     310 311 312
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
```
Comment (reneeotten) [inline python/py-uritools/Portfile:27]: dependency is implied as it is using "python.pep517 yes" with the default "setuptools"

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by catap** [comment]
@reneeotten thanks for review. @pmetzger I guess that CI had failed because of buildbots need some time to catch the lag. Now it should be better.

**Comment by reneeotten** [comment]
the CI has nothing to do with the buildbots - let's see how it goes this time.

**Comment by catap** [comment]
all good

**Comment by pmetzger** [comment]
Is this ready to merge?

**Comment by reneeotten** [comment]
> Is this ready to merge?

I'll take another look some time tonight or tomorrow and will merge when it is

**Review by reneeotten — APPROVED** [review]

**Comment by catap** [comment]
Thanks, Renee!

**Pull request merged**